exporter/stats/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"testing"
+
+	"go.opencensus.io/stats"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func newView(agg stats.Aggregation, window stats.Window) *stats.View {
+	m, _ := stats.NewMeasureInt64("tests/foo1", "bytes", "byte")
+	view, _ := stats.NewView("foo", "bar", nil, m, agg, window)
+	return view
+}
+
+type fauxWindow struct {
+	stats.Cumulative
+	name string
+}
+
+func TestOnlyCumulativeWindowSupported(t *testing.T) {
+	// See Issue https://github.com/census-instrumentation/opencensus-go/issues/214.
+	count1 := stats.CountData(1)
+	tests := []struct {
+		vds  *stats.ViewData
+		want int
+	}{
+		0: {
+			vds: &stats.ViewData{
+				View: newView(stats.CountAggregation{}, stats.Cumulative{}),
+			},
+			want: 0, // no rows present
+		},
+		1: {
+			vds: &stats.ViewData{
+				View: newView(stats.CountAggregation{}, stats.Cumulative{}),
+				Rows: []*stats.Row{
+					{nil, &count1},
+				},
+			},
+			want: 1,
+		},
+		2: {
+			vds: &stats.ViewData{
+				View: newView(stats.CountAggregation{}, fauxWindow{}),
```

Comment:
Just use Interval window instead.

Reply:
Nice, thanks. Updating that in a second.